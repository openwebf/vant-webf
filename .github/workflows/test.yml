name: CI

on:
  push:
    branches:
      - '**'

  pull_request:
    branches:
      - dev

  workflow_dispatch:

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run linter
      run: npm run lint

  test:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run test cases
      run: npm test

    - name: Upload coverage to Codecov  
      uses: codecov/codecov-action@v1
      with:
        token: ${{ secrets.CODECOV_TOKEN }}

  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Build
      run: npm run build
