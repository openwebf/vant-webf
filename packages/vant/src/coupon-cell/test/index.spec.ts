import { mount } from '../../../test';
import { CouponCell } from '../index';

const coupon = {
  id: 1,
  discount: 0,
  denominations: 150,
  originCondition: 0,
  reason: '',
  value: 150,
  name: 'name',
  description: 'description',
  startAt: 1489104000,
  endAt: 1514592000,
};

test('should render CouponCell correctly', () => {
  const wrapper = mount(CouponCell);
  expect(wrapper.html()).toMatchSnapshot();
});

test('should render CouponCell with chosenCoupon correctly', () => {
  const wrapper = mount(CouponCell, {
    props: {
      coupons: [coupon],
      chosenCoupon: 0,
    },
  });
  expect(wrapper.html()).toMatchSnapshot();
});

test('should render CouponCell correctly with zero discount', () => {
  const wrapper = mount(CouponCell, {
    props: {
      coupons: [{ ...coupon, value: 0, denominations: 150 }],
      chosenCoupon: 0,
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should click one time when coupon cell is clicked', () => {
  const onClick = jest.fn();
  const wrapper = mount(CouponCell, {
    props: {
      onClick,
    },
  });

  wrapper.trigger('click');
  expect(onClick).toHaveBeenCalledTimes(1);
});
