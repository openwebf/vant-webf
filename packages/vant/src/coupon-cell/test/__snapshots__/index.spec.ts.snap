// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CouponCell correctly 1`] = `
<div class="van-cell van-cell--clickable van-coupon-cell"
     role="button"
     tabindex="0"
>
  <div class="van-cell__title">
    <span>
      Coupon
    </span>
  </div>
  <div class="van-cell__value van-coupon-cell__value">
    <span>
      No coupons
    </span>
  </div>
  <i class="van-badge__wrapper van-icon van-icon-arrow van-cell__right-icon">
  </i>
</div>
`;

exports[`should render CouponCell correctly with zero discount 1`] = `
<div class="van-cell van-cell--clickable van-coupon-cell"
     role="button"
     tabindex="0"
>
  <div class="van-cell__title">
    <span>
      Coupon
    </span>
  </div>
  <div class="van-cell__value van-coupon-cell__value van-coupon-cell__value--selected">
    <span>
      -¥ 0.00
    </span>
  </div>
  <i class="van-badge__wrapper van-icon van-icon-arrow van-cell__right-icon">
  </i>
</div>
`;

exports[`should render CouponCell with chosenCoupon correctly 1`] = `
<div class="van-cell van-cell--clickable van-coupon-cell"
     role="button"
     tabindex="0"
>
  <div class="van-cell__title">
    <span>
      Coupon
    </span>
  </div>
  <div class="van-cell__value van-coupon-cell__value van-coupon-cell__value--selected">
    <span>
      -¥ 1.50
    </span>
  </div>
  <i class="van-badge__wrapper van-icon van-icon-arrow van-cell__right-icon">
  </i>
</div>
`;
